src/master-playlist-controller.js

```diff
@@ -408,6 +408,19 @@ export class MasterPlaylistController extends videojs.EventTarget {
         bubbles: true
       });
     });
+
+    this.masterPlaylistLoader_.on('playlistunchanged', () => {
+      let updatedPlaylist = this.masterPlaylistLoader_.media();
+      let playlistOutdated = this.stuckAtPlaylistEnd_(updatedPlaylist);
+
+      if (playlistOutdated) {
+        // Playlist has stopped updating and we're stuck at its end. Try to
+        // blacklist it and switch to another playlist in the hope that that
+        // one is updating (and give the player a chance to re-adjust to the
+        // safe live point).
+        this.blacklistCurrentPlaylist();
```

Comment:
We can also pass some error info into `blacklistCurrentPlaylist` regarding the error itself, so we can log appropriately.

e.g.,
```js
this.blacklistCurrentPlaylist({
  message: 'Playlist no longer updating.'
});
```

Then in `blacklistCurrentPlaylist` add a check for an error message and include it in the middle of logs (where it makes sense).